hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -355,12 +364,50 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
 
     (baseFileReader, preMergeBaseFileReader, skeletonReader, bootstrapBaseReader)
   }
+}
+
+object HoodieFileGroupReaderBasedParquetFileFormat {
+  private val ROW_INDEX = "row_index"
+  private val ROW_INDEX_TEMPORARY_COLUMN_NAME = s"_tmp_metadata_$ROW_INDEX"
+  private val FILE_SOURCE_GENERATED_METADATA_COL_ATTR_KEY = "__file_source_generated_metadata_col"
+  private val FILE_SOURCE_METADATA_COL_ATTR_KEY = "__file_source_metadata_col"
+  private val METADATA_COL_ATTR_KEY = "__metadata_col"
```

Comment:
Add docs on where these come from, i.e., related Spark classes and variables, and what they mean.